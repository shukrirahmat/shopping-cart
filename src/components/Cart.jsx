import styles from "./Cart.module.css";
import PropTypes, { func } from "prop-types";
import CartItem from "./CartItem";

function Cart(props) {

    function countOverall(array) {
        let total = 0;
        props.cartItems.forEach(item => {
            total += (item.price * item.amount);
        })
        return total.toFixed(2) + "$";
    }

  return (
    <div className={styles.wrapper}>
      <p className={styles.title}>Your Shopping Cart</p>
      {props.cartItems.length === 0 ? (
        <p className={styles.emptyText}>Your shopping cart is empty</p>
      ) : (
        <div className={styles.cartContainer}>
          {props.cartItems.map((item) => {
            return <CartItem item={item} key={item.id} />;
          })}
        </div>
      )}
      {props.cartItems.length === 0 ? null : (
        <div className={styles.overall}>
            <p>Overall Total : {countOverall()} </p>
          <button className={styles.checkout}>Checkout</button>
        </div>
      )}
    </div>
  );
}

Cart.propTypes = {
  cartItems: PropTypes.array,
};

export default Cart;
