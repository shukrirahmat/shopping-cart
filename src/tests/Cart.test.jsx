import Cart from "../components/Cart";
import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("cart component", () => {
    const item = {
        id: '69',
        title: 'test product',
        price: 13.5,
        description: 'lorem ipsum set',
        image: 'https://i.pravatar.cc',
        category: 'electronic',
        amount: 3
    }

    it("renders Cart page", () => {
        const container = render(<BrowserRouter><Cart cartItems={[item]} removeItem={() => {}} modifyAmount={() => {}}/></BrowserRouter>);
        expect(container).toMatchSnapshot();
    })

    it("draw empty message when no items", () => {
        render(<BrowserRouter><Cart cartItems={[]} removeItem={() => {}} modifyAmount={() => {}}/></BrowserRouter>);
        const emptyText = screen.queryByRole("paragraph", {name: /Your shopping cart is empty/i});
        expect(emptyText).toBeTruthy;
    })
})