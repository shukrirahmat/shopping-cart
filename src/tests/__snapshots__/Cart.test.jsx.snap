// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cart component > renders Cart page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_wrapper_c7a017"
      >
        <p
          class="_title_c7a017"
        >
          Your Shopping Cart
        </p>
        <div
          class="_cartContainer_c7a017"
        >
          <div
            class="_container_0d2c86"
          >
            <p
              class="_title_0d2c86"
            >
              test product
            </p>
            <div
              class="_imageContainer_0d2c86"
            >
              <img
                src="https://i.pravatar.cc"
              />
            </div>
            <div
              class="_amount_0d2c86"
            >
              <button>
                -
              </button>
              <input
                type="text"
                value="3"
              />
              <button>
                +
              </button>
            </div>
            <div
              class="_priceContainer_0d2c86"
            >
              <p>
                Price: 
                13.50$
              </p>
              <p>
                Total: 
                40.50$
              </p>
            </div>
            <button
              class="_removeButton_0d2c86"
            >
              Remove
            </button>
          </div>
        </div>
        <div
          class="_overall_c7a017"
        >
          <p>
            Overall Total : 
            40.50$
             
          </p>
          <button
            class="_checkout_c7a017"
          >
            <a
              class="_cotext_c7a017"
              href="/checkout"
            >
              Checkout
            </a>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_wrapper_c7a017"
    >
      <p
        class="_title_c7a017"
      >
        Your Shopping Cart
      </p>
      <div
        class="_cartContainer_c7a017"
      >
        <div
          class="_container_0d2c86"
        >
          <p
            class="_title_0d2c86"
          >
            test product
          </p>
          <div
            class="_imageContainer_0d2c86"
          >
            <img
              src="https://i.pravatar.cc"
            />
          </div>
          <div
            class="_amount_0d2c86"
          >
            <button>
              -
            </button>
            <input
              type="text"
              value="3"
            />
            <button>
              +
            </button>
          </div>
          <div
            class="_priceContainer_0d2c86"
          >
            <p>
              Price: 
              13.50$
            </p>
            <p>
              Total: 
              40.50$
            </p>
          </div>
          <button
            class="_removeButton_0d2c86"
          >
            Remove
          </button>
        </div>
      </div>
      <div
        class="_overall_c7a017"
      >
        <p>
          Overall Total : 
          40.50$
           
        </p>
        <button
          class="_checkout_c7a017"
        >
          <a
            class="_cotext_c7a017"
            href="/checkout"
          >
            Checkout
          </a>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
